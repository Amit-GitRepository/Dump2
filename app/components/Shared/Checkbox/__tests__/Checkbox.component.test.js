import CheckBox from '../Checkbox.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';

configure({adapter: new ReactSixteenAdapter()});

describe('CheckBox component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<CheckBox />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('shallow render for checkbox correctly', () => {
    const wrapper = shallow(<CheckBox />);
    expect(wrapper).toBeDefined();
  });
  it('should call onChange callback', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<CheckBox onChange={onChange}/>);
    wrapper.instance().onClick();
    expect(wrapper.instance().props.onChange).toBeCalledWith(true, '');
  });
});
