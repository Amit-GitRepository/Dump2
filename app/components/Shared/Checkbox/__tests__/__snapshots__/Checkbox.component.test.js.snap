// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Object {},
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#F93E60",
          "borderRadius": 5,
          "borderWidth": 2,
          "padding": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#B9C1C9",
            },
            Object {
              "fontFamily": "iservice",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        юдл
      </Text>
    </View>
  </View>
</View>
`;
